src/Telegram.Bot/TelegramBotClient.cs

```diff
@@ -1573,7 +1573,22 @@ public Task<Chat> GetChatAsync(ChatId chatId, CancellationToken cancellationToke
                 {"edit_message", editMessage}
             };
 
-            return SendWebRequestAsync<Message>("setGameScore", parameters, cancellationToken);
+            object response = await SendWebRequestAsync<object>("setGameScore", parameters, cancellationToken)
+                .ConfigureAwait(false);
+
+            Message message;
+            switch (response)
+            {
+                case Message m:
+                    message = m;
+                    break;
+                case bool b when b:
+                    message = default(Message);
+                    break;
+                default:
+                    throw new Exception($"Unexpected response: {response}");
+            }
+            return message;
```

Comment:
I don't believe any of those cases will actually happen. I expect tg to respond with error rather than `false` or any other unexpected value.